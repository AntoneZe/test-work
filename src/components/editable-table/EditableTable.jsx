import React, { useState } from "react";

import "./editable-table.less";

import PhoneEntryForm from "../phone-entry-form/PhoneEntryForm";
import Table from "../global/table/Table";
import InputSearch from "../global/input-search/InputSearch";
import Button from "../global/button/Button";
import Modal from "../global/modal/Modal";

const EditableTable = () => {
  const [modalActive, setModalActive] = useState(false);

  return (
    <>
      <div className="editable-table">
        <div className="editable-table__wrapper">
          <div className="editable-table__input-wrapper">
            <InputSearch />
          </div>
          <Button onClick={() => setModalActive(true)}>Добавить</Button>
        </div>
        <Table />
      </div>
      <Modal active={modalActive} setActive={setModalActive}>
        <PhoneEntryForm />
      </Modal>
    </>
  );
};

export default EditableTable;
