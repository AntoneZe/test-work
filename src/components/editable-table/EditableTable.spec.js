import React from "react";

import EditableTable from "./EditableTable";

describe("EditableTable component", () => {
  it("should render EditableTable component", () => {
    const component = render(
      <EditableTable
        titleList={["id", "name", "phone"]}
        rows={[{ id: 1, phone: "+79122142899", name: "Anton Zinchenko" }]}
        deleteCallback={() => {}}
        addCallback={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
