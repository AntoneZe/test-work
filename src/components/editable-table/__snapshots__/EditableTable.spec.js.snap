// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTable component should render EditableTable component 1`] = `
Array [
  <div
    class="editable-table"
  >
    <div
      class="editable-table__wrapper"
    >
      <div
        class="editable-table__input-wrapper"
      >
        <div
          class="custom-input-search"
        >
          <form>
            <div>
              <div
                class="custom-input-search__form-wrapper"
              >
                <input
                  id="mySearch"
                  name="q"
                  placeholder="Введите Имя..."
                  type="search"
                  value=""
                />
              </div>
              <button>
                <img
                  alt="альтернативный текст"
                  src="public/img/icons-search.svg"
                />
              </button>
            </div>
          </form>
        </div>
      </div>
      <button
        class="custom-button"
      >
        Добавить
      </button>
    </div>
    <div
      class="custom-table__wrapper"
    >
      <table
        class="custom-table"
      >
        <thead>
          <tr>
            <th>
              id
            </th>
            <th>
              name
            </th>
            <th>
              phone
            </th>
            <th />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td>
              Anton Zinchenko
            </td>
            <td>
              +79122142899
            </td>
            <td>
              <button
                class="custom-button"
              >
                <img
                  alt="delete"
                  height="15"
                  src="public/img/delete.svg"
                  width="15"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  <div
    class="custom-modal"
  >
    <div
      class="custom-modal__content"
    >
      <form>
        <fieldset
          class="custom-fieldset"
        >
          <legend>
            Добавить запись
          </legend>
          <div
            class="custom-fieldset__wrapper"
          >
            <label
              for="name"
            >
              Имя
            </label>
            <div
              class="custom-input__wrapper"
            >
              <input
                class="custom-input"
                id="name"
                maxlength="100"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="custom-fieldset__wrapper"
          >
            <label
              for="phone"
            >
              Номер телефона
            </label>
            <div
              class="custom-input__wrapper"
            >
              <input
                class="custom-input"
                id="phone"
                maxlength="18"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="custom-fieldset__wrapper custom-fieldset__wrapper--button"
          >
            <button
              class="custom-button"
              type="submit"
            >
              Добавить
            </button>
          </div>
        </fieldset>
      </form>
      <button
        class="custom-modal__close-button"
      />
    </div>
  </div>,
]
`;
